prow/pod-utils/gcs/upload.go

```diff
@@ -38,11 +39,11 @@ type destToWriter func(dest string) dataWriter
 // Upload uploads all of the data in the
 // uploadTargets map to GCS in parallel. The map is
 // keyed on GCS path under the bucket
-func Upload(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc) error {
+func Upload(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc, maxConcurrency int64) error {
 	dtw := func(dest string) dataWriter {
 		return gcsObjectWriter{bucket.Object(dest).NewWriter(context.Background())}
```

Comment:
This is kind of orthogonal to this PR, but we should be using a context with a timeout here so that we don't try to upload a file forever, especially now that that could prevent other files from being uploaded as well.